runtime/src/main/java/org/corfudb/runtime/object/transactions/OptimisticTransactionalContext.java

```diff
@@ -56,6 +49,13 @@
     private final Set<ICorfuSMRProxyInternal> modifiedProxies =
             new HashSet<>();
 
+    /** Keep the version (at global snapshot time)
```

Comment:
I don't know if this does that you want it to. The proxy is just a thin wrapper, the version is in the VersionLockedObject itself. Keeping versions would be quite expensive for large objects like maps.